transpiler/src/ast/core/BitloopsVisitor/helpers/valueObjectDeclaration.ts

```diff
@@ -20,26 +20,41 @@
 
 import BitloopsParser from '../../../../parser/core/grammar/BitloopsParser.js';
 import BitloopsVisitor from '../BitloopsVisitor.js';
-import { TValueObjectValues, TConstDeclarationValue } from '../../../../types.js';
+import { produceMetadata } from '../metadata.js';
+import { DomainCreateNode } from '../../intermediate-ast/nodes/Domain/DomainCreateNode.js';
+import { ConstDeclarationListNodeBuilder } from '../../intermediate-ast/builders/ConstDeclarationListBuilder.js';
+import { ConstDeclarationListNode } from '../../intermediate-ast/nodes/ConstDeclarationListNode.js';
+import { PrivateMethodDeclarationListNodeBuilder } from '../../intermediate-ast/builders/methods/PrivateMethodDeclarationListNodeBuilder.js';
+import { PrivateMethodDeclarationListNode } from '../../intermediate-ast/nodes/methods/PrivateMethodDeclarationListNode.js';
+import { ValueObjectIdentifierNodeBuilder } from '../../intermediate-ast/builders/valueObject/ValueObjectIdentifierNodeBuilder.js';
+import { ValueObjectDeclarationNodeBuilder } from '../../intermediate-ast/builders/valueObject/ValueObjectDeclarationNodeBuilder.js';
 
 export const valueObjectDeclarationVisitor = (
   thisVisitor: BitloopsVisitor,
   ctx: BitloopsParser.ValueObjectDeclarationContext,
-): { ValueObjects: { [identifier: string]: TValueObjectValues } } => {
+): void => {
+  const metadata = produceMetadata(ctx, thisVisitor);
+
   const valueObjectIdentifier = ctx.valueObjectIdentifier().getText();
-  const domainConstructorDeclaration = thisVisitor.visit(ctx.domainConstructorDeclaration());
-  const constantVars: TConstDeclarationValue[] = thisVisitor.visit(
-    ctx.domainConstDeclarationList(),
+  const valueObjectIdentifierNode = new ValueObjectIdentifierNodeBuilder(metadata)
```

Comment:
I think metadata here will be for the whole value object, and not just the identifier?

Reply:
Oh yes you are right! So maybe I have to create a visitor for the identifier??